### PR TITLE
[v2] docs - update page query docs

### LINKED ISSUE #7278: [v2] docs - update page query docs
branch off #7231

Update tutorial and docs -- move "How does the graphql tag work?" section to page query docs, out of tutorial (https://github.com/gatsbyjs/gatsby/pull/7240#issuecomment-412502818)

### PR DESCRIPTION
**Pull request by amberleyromo**
closes #7278

Changes:

**docs/build-a-page-with-graphql-query**
- worked toward consistency with the static query docs
  - renamed this page to docs/page-query. These docs were tucked away in an "example" style page; for documentation (as opposed to tutorial style), I thought it made a lot more sense to have the page-query/static-query documentation be patterned similarly.
 
**docs/static-query**
- slight edits for consistency with `docs/page-query` docs.

**sidebars/doc-links**
- reordered to reflect the change in architecture: creating pages, then page query, then StaticQuery

**tutorial/part-four**
- a la @m-allanson https://github.com/gatsbyjs/gatsby/pull/7240#issuecomment-412502818: removed the "How does the `graphql` tag work?" section and integrated into `docs/page-query` 
- added a link back to `docs/page-query` (`docs/static-query` is already linked)

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview for *using-postcss-sass* failed.

Built with commit 85f13da220bce5104449cbe72a2c5290bf6fe42f

https://app.netlify.com/sites/using-postcss-sass/deploys/5b71d1d7dd28ef1b1bd94629

**Comment by KyleAMathews** [comment]
Deploy preview for *using-contentful* failed.

Built with commit 85f13da220bce5104449cbe72a2c5290bf6fe42f

https://app.netlify.com/sites/using-contentful/deploys/5b71d1d8dd28ef1b1bd94638

**Comment by KyleAMathews** [comment]
Deploy preview for *using-jss* failed.

Built with commit 85f13da220bce5104449cbe72a2c5290bf6fe42f

https://app.netlify.com/sites/using-jss/deploys/5b71d1dbdd28ef1b1bd94650

**Comment by KyleAMathews** [comment]
Deploy preview for *gatsbyjs* failed.

Built with commit 85f13da220bce5104449cbe72a2c5290bf6fe42f

https://app.netlify.com/sites/gatsbyjs/deploys/5b71d1d6dd28ef1b1bd9461d

File: docs/docs/static-query.md [hunk 1]
```diff
@@ -1,10 +1,14 @@
 ---
-title: "Querying data in components using StaticQuery"
+title: Querying data in components using StaticQuery
 ---
 
 Gatsby v2 introduces `StaticQuery`, a new API that allows components to retrieve data via GraphQL query.
 
-## Basic example
+In this guide, we'll walk through an example using `StaticQuery`, and discuss [the difference between a StaticQuery and a page query](/static-query/#how-staticquery-differs-from-page-query).
```
Comment (jlengstorf) [inline docs/docs/static-query.md]: This is going to end up with a merge conflict in #7249, but otherwise I'm into it. 😄

**Pull request merged**